src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,174 @@
+    public static final String SPACER = "  ";
+
+    private final String filePath;
+    public Command(String command, TaskList tasks, String filePath) {
+        this.command = command;
+        this.tasks = tasks;
+        this.filePath = filePath;
+    }
+
+    private static void markCommand(String input, TaskList tasks) throws DukeException {
+        String[] words = input.split(" ");
+        int num = Integer.parseInt(words[1]);
+        if (num <= (tasks.getSize())) {
+            tasks.findTask(num - 1).markDone();
+            System.out.println("that was fast\n" + SPACER + tasks.findTask(num - 1).toString() + System.lineSeparator()
+                    + DIVIDER);
+        } else {
+            throw new DukeException();
+        }
+    }
+
+    private static void unmarkCommand(String input, TaskList tasks) throws DukeException {
+        String[] words = input.split(" ");
+        int num = Integer.parseInt(words[1]);
+        if (num <= (tasks.getSize())) {
+            tasks.findTask(num - 1).markUndone();
+            System.out.println("can you make up your mind\n" + SPACER + tasks.findTask(num - 1).toString() + System.lineSeparator()
+                    + DIVIDER);
+        } else {
+            throw new DukeException();
+        }
+    }
+
+    private static Deadline deadlineCommand(String input) throws DukeException {
+        if (input.length() < 10 || !input.contains("/")) {
+            throw new DukeException();
+        }
+        int endOfDescription = input.indexOf("/") - 1;
+        String description = input.substring(9, endOfDescription);
+        String by = input.substring(input.indexOf("/") + 1);
+        return new Deadline(description, by);
+    }
+
+    private static Todo todoCommand(String input) throws DukeException {
+        if (input.length() < 6) {
+            throw new DukeException();
+        }
+        String description = input.substring(5);
+        return new Todo(description);
+    }
+
+    private static Event eventCommand(String input) throws DukeException {
+        if (input.length() < 7 || !input.contains("/")) {
+            throw new DukeException();
+        }
+        int endOfDescription = input.indexOf("/") - 1;
+        String description = input.substring(6, endOfDescription);
+        String at = input.substring(input.indexOf("/") + 1);
+        return new Event(description, at);
+    }
+
+    private static void deleteCommand(TaskList tasks, String command) throws DukeException {
+        String[] words = command.split(" ");
+        int num = Integer.parseInt(words[1]);
+        if (num <= tasks.getSize()) {
+            System.out.println("task deleted" + System.lineSeparator() + SPACER + tasks.findTask(num - 1).toString()
+                    + System.lineSeparator() + "you still have " + (tasks.getSize() - 1) + " tasks left" + System.lineSeparator()
+                    + DIVIDER);
```

Comment:
Consider splitting this statement by storing some of the function returns in variables to allow readability. Similar suggestions apply to other lines as well.